kore/src/Kore/Logger/Output.hs

```diff
@@ -223,20 +229,26 @@ runLoggerT :: KoreLogOptions -> LoggerT IO a -> IO a
 runLoggerT options loggerT = do
     let runLogger = runReaderT . getLoggerT $ loggerT
     withLogger options $ \logger -> do
-        modifiedLogger <- concurrentLogger logger
-        runLogger modifiedLogger
+        (asyncThread, modifiedLogger) <- concurrentLogger logger
+        result <- runLogger modifiedLogger
+        wait asyncThread
```

Comment:
We should use [`finally`](https://hackage.haskell.org/package/exceptions-0.10.4/docs/Control-Monad-Catch.html#v:finally) so that if `runLogger modifiedLogger` throws an exception, the logging thread gets a chance to finish writing.